codalab/server/bundle_manager.py

```diff
@@ -329,29 +335,29 @@ def _schedule_run_bundles_on_workers(self, workers, staged_bundles_to_run):
                 # Get all the CodaLab's public workers
                 workers_list = workers.user_owned_workers(self._model.root_user_id)
 
-            workers_list = self._deduct_worker_resources(workers_list)
+            workers_list = self._deduct_worker_resources(workers_list, uuid_to_running_bundles)
             workers_list = self._filter_and_sort_workers(workers_list, bundle, bundle_resources)
 
             # Try starting bundles on the workers that have enough computing resources
             for worker in workers_list:
                 if self._try_start_bundle(workers, worker, bundle):
                     break
 
-    def _deduct_worker_resources(self, workers_list):
+    def _deduct_worker_resources(self, workers_list, uuid_to_running_bundles):
         """
         From each worker, subtract resources used by running bundles. Modifies the list.
         """
+
         for worker in workers_list:
             for uuid in worker['run_uuids']:
-                try:
-                    bundle = self._model.get_bundle(uuid)
-                except NotFoundError:
+                # Verify if the current bundle exists in both Worker table and Bundle table
```

Comment:
fix grammar/terminology: "the `worker` table and the `bundle` table" (comment applies throughout)

Reply:
This database call takes around 50% to 70% of the total time for each iteration. So this change should be able to help a bit when dispatching jobs.